spark/src/main/scala/geotrellis/spark/SpatialKey.scala

```diff
@@ -16,12 +16,17 @@
 
 package geotrellis.spark
 
+import geotrellis.spark.tiling.LayoutDefinition
+import geotrellis.vector.Extent
 import org.apache.spark.rdd.RDD
 
 /** A SpatialKey designates the spatial positioning of a layer's tile. */
 case class SpatialKey(col: Int, row: Int) extends Product2[Int, Int] {
   def _1 = col
   def _2 = row
+
+  /** Retrieve the [[Extent]] that corresponds to this key, given a layout. */
+  def tileExtent(layout: LayoutDefinition): Extent = layout.mapTransform(this)
```

Comment:
The reference to `tile` here makes this key, which is agnostic of Value type, seem like it's specific to `Tile`. We've been avoiding placing specific language on things that could be generally useful, e.g. to point clouds that are stored based on spatial keys.

Reply:
Hmm, thats tough. `SpatialKey` covers a rectangular, tile-like area on the map based on `LayoutDefinition`.